library/src/updater.rs

```diff
@@ -805,4 +834,45 @@ mod tests {
         })
         .unwrap();
     }
+
+    #[test]
+    fn no_config_lock_contention_when_waiting_for_patch_check() {
+        let tmp_dir = TempDir::new("example").unwrap();
+        init_for_testing(&tmp_dir, None);
+
+        // Set up the network hooks to sleep for 10 seconds on a patch check request
+        let hooks = NetworkHooks {
+            patch_check_request_fn: |_url, _request| {
+                let patch_check_delay = std::time::Duration::from_secs(2);
+                std::thread::sleep(patch_check_delay);
+                // If we get here, the test has failed.
+                unreachable!("If the test has not terminated before this, set_config is likely being blocked by a patch check request, which should not happen");
+            },
+            download_file_fn: |_url| Ok([].to_vec()),
+            report_event_fn: |_url, _event| Ok(()),
+        };
+
+        testing_set_network_hooks(
+            hooks.patch_check_request_fn,
+            hooks.download_file_fn,
+            hooks.report_event_fn,
+        );
+
+        // Invoke check_for_update to kick off a patch check request
+        let _ = std::thread::spawn(|| {
+            crate::check_for_update().unwrap();
```

Comment:
I think you want .ok() instead of .unwrap().  Unwrap will just ignore the error, right?  Where as ok() will panick if it errored?

Reply:
Apologies for swapping the two. Thanks!